### PR TITLE
84099 1010 ezr add gender to post fill

### PR DESCRIPTION
**Pull request by JoshingYou1**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES/NO*
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/form1010_ezr/service.rb [hunk 1]
```diff
@@ -153,10 +150,20 @@ def post_fill_veteran_ssn(parsed_form)
...
 
+    def post_fill_veteran_gender(parsed_form)
+      return if parsed_form['gender'].present?
+
+      StatsD.increment("#{Form1010Ezr::Service::STATSD_KEY_PREFIX}.missing_gender")
+
+      parsed_form['gender'] = @user.gender
+      parsed_form
+    end
+
```
Comment (Mitch-A6) [inline lib/form1010_ezr/service.rb:161]: 🎨 Not to echo CodeClimate here, but with this many postfilled fields now, would it make sense to abstract this pattern to a more general `post_fill_field(parsed_form, field_name)` or something?

If we're expecting this code to all go away soon™ it might not be worth it, but this debugging has been going on for a fair bit already....

**Review by Mitch-A6 — APPROVED** [review]
LGTM - comment is more of a thought question, and is not blocking.

File: lib/form1010_ezr/service.rb [hunk 2]
```diff
@@ -153,10 +150,20 @@ def post_fill_veteran_ssn(parsed_form)
       parsed_form
     end
 
+    def post_fill_veteran_gender(parsed_form)
+      return if parsed_form['gender'].present?
+
+      StatsD.increment("#{Form1010Ezr::Service::STATSD_KEY_PREFIX}.missing_gender")
+
+      parsed_form['gender'] = @user.gender
+      parsed_form
```
Comment (coope93) [inline lib/form1010_ezr/service.rb:159]: I know we do this on the other post-fill logic, but why are we returning the `parsed_form` from each of these methods if we are already mutating the passed in `parsed_form`?  I see that helps with testing the specific methods, but ultimately it makes it really hard to follow where all we are mutating our `parsed_form` before actually submitting it. Do we like that pattern? I know for me it makes it hard to follow and ultimately more error prone.

File: lib/form1010_ezr/service.rb (hunk 1 above)
Comment (JoshingYou1) [reply lib/form1010_ezr/service.rb]: Yes, this makes sense! I cleaned up the code, so please feel free to re-review!

File: lib/form1010_ezr/service.rb (hunk 2 above)
Comment (JoshingYou1) [reply lib/form1010_ezr/service.rb]: I can understand the confusion. I refactored the code so we aren't returning `parsed_form` multiple times. Please feel free to re-review!

**Comment by coope93** [comment]
I like the refactor. I think it simplifies it a lot, and I like the array with the user fields and how you implemented the loop to add them vs having a new method for each one. Nice work on this one!

**Review by coope93 — APPROVED** [review]

File: lib/form1010_ezr/service.rb [hunk 3]
```diff
@@ -120,48 +121,29 @@ def post_fill_required_fields(parsed_form)
...
+      required_user_form_fields = {
+        'veteranDateOfBirth' => @user.birth_date,
+        'veteranFullName' => @user.full_name_normalized&.compact&.stringify_keys,
+        'veteranSocialSecurityNumber' => @user.ssn_normalized,
+        'gender' => @user.gender
+      }
 
-      StatsD.increment("#{Form1010Ezr::Service::STATSD_KEY_PREFIX}.missing_date_of_birth")
+      required_user_form_fields.each do |key, value|
+        next if parsed_form[key.to_s].present?
```
Comment (Mitch-A6) [inline lib/form1010_ezr/service.rb:136]: 🤔 Do we need the `to_s` here given the `key`s are defined as strings above?
Alternatively, how would you feel about keeping the `to_s` and changing the keys to symbols for the hash definition shorthand above? (e.g., `veteranDateOfBirth: @user.birth_date,`)

Regardless, this wound up a lot cleaner - nice!

**Review by Mitch-A6 — DISMISSED** [review]

File: lib/form1010_ezr/service.rb (hunk 3 above)
Comment (JoshingYou1) [reply lib/form1010_ezr/service.rb:136]: I think making the keys symbols would be an issue because the hash is in JSON format. Finding `parsed_form[:veteranDateOfBirth]` would result in a `nil` value.

**Review by Mitch-A6 — APPROVED** [review]
👍

**Review by coope93 — APPROVED** [review]
